### ISSUE TITLE
Slow initialization phase on frontera when getting closer to full machine

### ISSUE THREAD

**Issue by Thomas-Ulrich** [SeisSol/SeisSol#840]
**Describe the bug**
One specific part of the initialization increases a lot when using more nodes.
E.g.

2000 nodes: 52s
(init until there: 2min 10)
```
Thu Apr 13 05:36:53, Info:  Initializing energy output.
Rank:        0 | Info    | <--------------------------------------------------------->
Rank:        0 | Info    | <     Start calc_SeisSol ...                              >
Rank:        0 | Info    | <--------------------------------------------------------->
Thu Apr 13 05:37:45, Info:  Writing faultoutput at time 0.
```
(/work2/09160/ulrich/frontera/Turkey-Syria-Earthquakes/SeisSolSetupHeterogeneities/event2/5397156_470M_2000.out)

4000 nodes: 3min 28
(init until there: 2min 38)
```
Thu Apr 13 05:53:22, Info:  Initializing energy output.
Rank:        0 | Info    | <--------------------------------------------------------->
Rank:        0 | Info    | <     Start calc_SeisSol ...                              >
Rank:        0 | Info    | <--------------------------------------------------------->
Thu Apr 13 05:56:50, Info:  Writing faultoutput at time 0.
````
(/work2/09160/ulrich/frontera/Turkey-Syria-Earthquakes/SeisSolSetupHeterogeneities/event2/5397158_470M_4000.out)
and 8000 nodes: 18min 51s
(init until there: 4min 18)
```
Thu Apr 13 06:08:10, Info:  Initializing energy output.
Rank:        0 | Info    | <--------------------------------------------------------->
Rank:        0 | Info    | <     Start calc_SeisSol ...                              >
Rank:        0 | Info    | <--------------------------------------------------------->
Thu Apr 13 06:27:01, Info:  Writing faultoutput at time 0.
```
(/work2/09160/ulrich/frontera/Turkey-Syria-Earthquakes/SeisSolSetupHeterogeneities/event2/5397161_470M_8000.out)

**Expected behavior**
no node dependence.

**To Reproduce**
1. Which version do you use? Provide branch and commit id.
master, 8d6e4551ea1a070b5b7d8261582f2b532e63dccd
2. Which build settings do you use? Which compiler version do you use? 
intel 
```
ADDRESS_SANITIZER_DEBUG          OFF
 ASAGI                            ON
 CMAKE_BUILD_TYPE                 Release
 CMAKE_INSTALL_PREFIX             /usr/local
 COMMTHREAD                       ON
 COVERAGE                         OFF
 DEVICE_ARCH                      none
 DEVICE_BACKEND                   none
 DR_QUAD_RULE                     dunavant
 EQUATIONS                        viscoelastic2
 Eigen3_DIR                       /work2/09160/ulrich/frontera/spack/opt/spack/linux-centos7-cascadelake/intel-19.1.1.217/eigen-3.4.0-umuh726eijyrt346dic275mdha6rc7mm/share/eigen3/cmake
 GEMM_TOOLS_LIST                  LIBXSMM,PSpaMM
 HDF5                             ON
 HDF5_DIR                         /work2/09160/ulrich/frontera/spack/opt/spack/linux-centos7-cascadelake/intel-19.1.1.217/hdf5-1.12.2-fzowvyobpiowc3f3hsidtdj3mkxpkm6x/cmake
 HOST_ARCH                        skx
 INTEGRATE_QUANTITIES             OFF
 LIKWID                           OFF
 LOG_LEVEL                        warning
 LOG_LEVEL_MASTER                 info
 Libxsmm_executable_PROGRAM       /work2/09160/ulrich/frontera/spack/opt/spack/linux-centos7-cascadelake/intel-19.1.1.217/libxsmm-1.17-qdddzbw5bx5r6lkzihnuu44ry5uyaavv/bin/libxsmm_gemm_generator
 MEMKIND                          OFF
 MEMORY_LAYOUT                    auto
 METIS                            ON
 MINI_SEISSOL                     ON
 MPI                              ON
 NETCDF                           ON
 NUMA_AWARE_PINNING               ON
 NUMA_ROOT_DIR                    /usr
 NUMBER_OF_FUSED_SIMULATIONS      1
 NUMBER_OF_MECHANISMS             3
 OPENMP                           ON
 ORDER                            6
 PLASTICITY_METHOD                nb
 PRECISION                        double
 PROXY_PYBINDING                  OFF
 PSpaMM_PROGRAM                   /work2/09160/ulrich/frontera/spack/opt/spack/linux-centos7-cascadelake/intel-19.1.1.217/py-pspamm-develop-35zum726336t2gdzfcz6mfzivifra7nd/pspamm.py
 SIONLIB                          OFF
 TESTING                          OFF
 TESTING_GENERATED                OFF
 USE_IMPALA_JIT_LLVM              OFF
 easi_DIR                         /work2/09160/ulrich/frontera/spack/opt/spack/linux-centos7-cascadelake/intel-19.1.1.217/easi-1.2.0-zto7che54ra7wvybw2be6m66yzaisxt5/lib64/cmake/easi
 impalajit_DIR                    /work2/09160/ulrich/frontera/spack/opt/spack/linux-centos7-cascadelake/intel-19.1.1.217/impalajit-main-rdjaykqjjbb645iny6nexrtnup27ejpg/lib64/cmake/impalajit
 netCDF_DIR                       netCDF_DIR-NOTFOUND
 yaml-cpp_DIR                     /work2/09160/ulrich/frontera/spack/opt/spack/linux-centos7-cascadelake/intel-19.1.1.217/yaml-cpp-0.6.2-qszzfashukprv326kpoe2ivdwrfq6a5f/lib/cmake/yaml-cpp
```

3. On which machine does your problem occur? If on a cluster: Which modules are loaded?
frontera, 
```
Currently Loaded Modules:
  1) intel/19.1.1    7) hwloc/1.11.12
  2) impi/19.0.9     8) xalt/2.10.34
  3) git/2.24.1      9) TACC
  4) autotools/1.2  10) python3/3.9.2
  5) cmake/3.24.2   11) seissol-env-develop-intel-19.1.1.217-w2i565p
  6) pmix/3.1.4     12) impalajit-main-intel-19.1.1.217-rdjaykq
```

5. Provide parameter/material files.
/work2/09160/ulrich/frontera/Turkey-Syria-Earthquakes/SeisSolSetupHeterogeneities/event2

**Comment by krenzland** [comment]
Just to clarify, it is NOT related to the energy output. It happens also without.
I'm not 100% certain that this is actually a bug, it could also be caused by some MPI issue on Frontera.

**Comment by wangyinz** [comment]
Another thing worth mentioning is that when I login to some the compute nodes, I can see the SeisSol processes are idling with almost no CPU usage.

**Comment by Thomas-Ulrich** [comment]
7min30 now to initialize the 685M cell mesh!
```
ulrich@login1:/scratch1/09160/ulrich/Turkey-Syria-Earthquakes/SeisSolSetupHeterogeneities/event2$ cat 5861721.out 

Currently Loaded Modules:
  1) intel/19.1.1    8) xalt/2.10.34
  2) impi/19.0.9     9) TACC
  3) git/2.24.1     10) python-3.9.2-intel-19.1.1.217-gi5my2a
  4) autotools/1.2  11) python3/3.7.0
  5) cmake/3.24.2   12) hdf5-1.10.7-intel-19.1.1.217-pxgykdc
  6) pmix/3.1.4     13) phdf5/1.10.4
  7) hwloc/1.11.12

 

/scratch1/09160/ulrich/Turkey-Syria-Earthquakes/SeisSolSetupHeterogeneities/event2
Do 12. Okt 14:03:24 CDT 2023
num_nodes: 8165 ntasks: 16330
/scratch1/09160/ulrich/SeisSol/build/SeisSol_Release_dskx_6_viscoelastic2
TACC:  Starting up job 5861721 
TACC:  Starting parallel tasks... 
[0] MPI startup(): I_MPI_EXTRA_FILESYSTEM_LIST environment variable is not supported.
[0] MPI startup(): Similar variables:
         I_MPI_EXTRA_FILESYSTEM
         I_MPI_EXTRA_FILESYSTEM_FORCE
[0] MPI startup(): To check the list of supported variables, use the impi_info utility or refer to https://software.intel.com/en-us/mpi-library/documentation/get-started.
Thu Oct 12 14:05:02, Info:  Welcome to SeisSol 
Thu Oct 12 14:05:02, Info:  Copyright (c) 2012 - 2023  SeisSol Group 
Thu Oct 12 14:05:02, Info:  Version: v1.1.0-36-g4b7c3f33 (dirty) 
Thu Oct 12 14:05:02, Info:  Built on: Oct 12 2023 08:36:04 
Thu Oct 12 14:05:02, Info:  Last commit: 4b7c3f335e67a59cc4e07bd5bd2f11c2f59e4b3c at 2023-10-12T12:22:04+00:00 
Thu Oct 12 14:05:02, Info:  Running on: "c101-001.frontera.tacc.utexas.edu" 
Thu Oct 12 14:05:02, Info:  Using MPI with #ranks: 16330 
Thu Oct 12 14:05:02, Info:  Using OMP with #threads/rank: 27 
Thu Oct 12 14:05:02, Info:  OpenMP worker affinity (this process): "0-2-4-6-8-|0-2-4-6-8-|0-2-4-6-8-|0-2-4-6-8-|0---4-6-8-|0-2-4-" 
Thu Oct 12 14:05:02, Info:  OpenMP worker affinity (this node)   : "0123456789|0123456789|0123456789|0123456789|01--456789|012345" 
Thu Oct 12 14:05:02, Info:  Communication thread affinity        : "----------|----------|----------|----------|--2-------|------" 
Thu Oct 12 14:05:02, Info:  The stack size ulimit is  2097152 [kb]. 
Thu Oct 12 14:05:03, Info:  Reading SeisSol parameter file... 
Thu Oct 12 14:05:03, Info:  Setting maximum timestep width to 0.0357143  for visco-elastic material (as specified in the documentation). 
Thu Oct 12 14:05:03, Info:  In your parameter file, you have specified a non-positive interval for "checkpointinterval" . This mechanism is deprecated and may be removed in a future version of SeisSol. Consider disabling the whole module by setting "checkpoint" to 0 instead by adding "checkpoint = 0" to the "output" section of your parameter file instead. 
Thu Oct 12 14:05:03, Info:  The field "faultoutputflag" is no longer in use. You may safely remove it from your parameters file. 
Thu Oct 12 14:05:03, Info:  The field "boundaries" is no longer in use. You may safely remove it from your parameters file. 
Thu Oct 12 14:05:03, Info:  SeisSol parameter file read successfully. 
Thu Oct 12 14:05:03, Info:  Model information: 
Thu Oct 12 14:05:03, Info:  Elastic model: no 
Thu Oct 12 14:05:03, Info:  Viscoelastic model: yes 
Thu Oct 12 14:05:03, Info:  Anelastic model: yes 
Thu Oct 12 14:05:03, Info:  Poroelastic model: no 
Thu Oct 12 14:05:03, Info:  Anisotropic model: no 
Thu Oct 12 14:05:03, Info:  Plasticity: yes 
Thu Oct 12 14:05:03, Info:  Begin init mesh. 
Thu Oct 12 14:05:03, Info:  Mesh file: "mesh/Turkey_extended_13" 
Thu Oct 12 14:05:03, Info:  Running mini SeisSol to determine node weight 
Thu Oct 12 14:05:03, Info:  miniSeisSol configured with 50000 elements and 10 repeats per process 
Thu Oct 12 14:05:03, Info:  Node weights: mean = 2.05101  std = 0.0112421  min = 1.94138  median = 2.05427  max = 2.06231 
Thu Oct 12 14:05:03, Info:  Reading PUML mesh 
Thu Oct 12 14:05:08, Info:  Found 684983574 cells 
Thu Oct 12 14:05:17, Info:  Found 115877698 vertices 
Thu Oct 12 14:05:30, Info:  Computing LTS weights. 
Thu Oct 12 14:05:30, Warn:  Material Averaging is not implemented for plastic materials. Falling back to material properties sampled from the element barycenters instead. 
Thu Oct 12 14:07:43, Info:  Using best wiggle factor as baseline cost for auto merging. 
Thu Oct 12 14:07:43, Info:  1. Compute best wiggle factor without merging clusters 
Thu Oct 12 14:07:43, Info:  Baseline cost, without wiggle factor and cluster merging is 5.43176e+13 
Thu Oct 12 14:07:43, Info:  Enforcing maximum difference when finding best wiggle factor took 0 reductions. 
Thu Oct 12 14:07:43, Info:  The best wiggle factor is 0.91 with cost 5.06139e+13 and 8 time clusters 
Thu Oct 12 14:07:43, Info:  Cost decreased 6.81864 % with absolute cost decrease of 3.70372e+12 compared to the baseline 
Thu Oct 12 14:07:43, Info:  2. Compute best wiggle factor with merging clusters, using the previous cost estimate as baseline 
Thu Oct 12 14:07:43, Info:  Baseline cost before cluster merging is 5.06139e+13 
Thu Oct 12 14:07:43, Info:  Maximal admissible cost after cluster merging is 6.07366e+13 
Thu Oct 12 14:07:43, Info:  Enforcing maximum difference when finding best wiggle factor took 0 reductions. 
Thu Oct 12 14:07:43, Info:  The best wiggle factor is 0.91 with cost 5.47144e+13 and 4 time clusters 
Thu Oct 12 14:07:43, Info:  Cost increased 8.10156 % with absolute cost increase of 4.10051e+12 compared to the baseline 
Thu Oct 12 14:07:43, Info:  Note: Cost increased due to cluster merging! 
Thu Oct 12 14:07:43, Info:  Limiting number of clusters to 4 
Thu Oct 12 14:07:43, Info:  Computing LTS weights. Done.  (175482 reductions.)
Thu Oct 12 14:07:43, Info:  Using the Parmetis partition library and strategy. 
Thu Oct 12 14:07:52, Info:  PUML mesh read in: 168.649 (min: 168.587, max: 168.833)
Thu Oct 12 14:07:52, Info:  The mesh has been read. Starting post processing. 
Thu Oct 12 14:07:52, Info:  Extracting fault information. 
Thu Oct 12 14:07:52, Info:  Exchanging ghostlayer metadata. 
Thu Oct 12 14:07:52, Warn:  Material Averaging is not implemented for plastic materials. Falling back to material properties sampled from the element barycenters instead. 
Thu Oct 12 14:08:11, Info:  Mesh initialized in: 188.176 (min: 188.156, max: 188.36)
Thu Oct 12 14:08:11, Info:  End init mesh. 
Thu Oct 12 14:08:11, Info:  Begin init model. 
Thu Oct 12 14:08:11, Info:  Initialize LTS. 
Thu Oct 12 14:08:11, Info:  Deriving clusters ids for min. time step width / multiRate: 7.39218e-05 / 2 
Thu Oct 12 14:08:11, Info:  Due to wiggle factor of 0.91 the minimum timestep size is reduced to 6.72688e-05 
Thu Oct 12 14:08:11, Info:  Number of elements in time clusters: 
Thu Oct 12 14:08:11, Info:  0: 3446 
Thu Oct 12 14:08:11, Info:  1: 318063 
Thu Oct 12 14:08:11, Info:  2: 48249262 
Thu Oct 12 14:08:11, Info:  3: 636412803 
Thu Oct 12 14:08:11, Info:  maximum theoretical speedup (compared to GTS): 9.28613 per cell LTS, 6.79189 with the used clustering. 
Thu Oct 12 14:08:11, Info:  Number of elements in dynamic rupture time clusters: 
Thu Oct 12 14:08:11, Info:  0 (dr): 67 
Thu Oct 12 14:08:11, Info:  1 (dr): 11592 
Thu Oct 12 14:08:11, Info:  2 (dr): 522420 
Thu Oct 12 14:08:11, Info:  3 (dr): 28613 
Thu Oct 12 14:08:11, Info:  Initialize Friction Model 
Thu Oct 12 14:08:29, Info:  Initialize cell material parameters. 
Thu Oct 12 14:08:29, Warn:  Material Averaging is not implemented for plastic materials. Falling back to material properties sampled from the element barycenters instead. 
Thu Oct 12 14:09:15, Warn:  Material Averaging is not implemented for plastic materials. Falling back to material properties sampled from the element barycenters instead. 
Thu Oct 12 14:09:33, Info:  Initialize Memory layout. 
Thu Oct 12 14:09:44, Info:  Initialize cell-local matrices. 
Thu Oct 12 14:09:44, Info:  Initializing Fault, using a quadrature rule with  33  points. 
Thu Oct 12 14:12:16, Info:  Model initialized in: 245.071 (min: 245.068, max: 245.075)
Thu Oct 12 14:12:16, Info:  End init model. 
Thu Oct 12 14:12:16, Info:  Setting initial conditions. 
Thu Oct 12 14:12:16, Info:  Using initial condition "Zero" . 
Thu Oct 12 14:12:16, Info:  Reading source. 
Thu Oct 12 14:12:16, Info:  No source term specified. 
Thu Oct 12 14:12:16, Info:  Setting up boundary conditions. 
Thu Oct 12 14:12:16, Info:  Begin init output. 
Thu Oct 12 14:12:16, Info:  Initializing free surface integrator. 
Thu Oct 12 14:12:16, Info:  Initializing free surface integrator. Done. 
Thu Oct 12 14:12:16, Info:  Enabling 2D fault output 
Thu Oct 12 14:12:16, Info:  Initializing Fault output. Number of sub-triangles: 4 
Thu Oct 12 14:12:16, Info:  Initializing XDMF fault output. 
Thu Oct 12 14:12:17, Info:  Fault writer thread affinity: "----------|----------|----------|----------|--2-------|------" 
Thu Oct 12 14:12:20, Info:  Initializing XDMF fault output. Done. 
Thu Oct 12 14:12:20, Info:  Wave field writer thread affinity: "----------|----------|----------|----------|--2-------|------" 
Thu Oct 12 14:12:20, Info:  Initializing XDMF wave field output. 
Thu Oct 12 14:12:20, Info:  Refinement is turned off. 
Thu Oct 12 14:12:20, Info:  Refinement class initialized 
Thu Oct 12 14:12:20, Info:  VariableSubsampler initialized 
Thu Oct 12 14:12:26, Info:  High order output initialized 
Thu Oct 12 14:12:26, Info:  Initializing XDMF wave field output. Done. 
Thu Oct 12 14:12:26, Info:  Initializing free surface output. 
Thu Oct 12 14:12:27, Info:  Free surface writer thread affinity: "----------|----------|----------|----------|--2-------|------" 
Thu Oct 12 14:12:27, Info:  Record points read from "mesh/Turkey_extended_13_stations_proj_m5.txt" 
Thu Oct 12 14:12:27, Info:  Number of record points = 484 
Thu Oct 12 14:12:27, Info:  Finding meshIds for receivers... 
Thu Oct 12 14:12:27, Info:  Cleaning possible double occurring receivers for MPI... 
Thu Oct 12 14:12:29, Info:  Initializing free surface output. Done. 
Thu Oct 12 14:12:29, Info:  Mapping receivers to LTS cells... 
Thu Oct 12 14:12:29, Info:  Initializing energy output. 
Thu Oct 12 14:12:29, Info:  End init output. 
Thu Oct 12 14:12:38, Info:  Finishing initialization... 
Thu Oct 12 14:12:38, Info:  Starting simulation. 
Thu Oct 12 14:12:38, Info:  Writing faultoutput at time 0.
Thu Oct 12 14:12:38, Info:  Writing faultoutput at time 0. Done.
Thu Oct 12 14:12:38, Info:  Waiting for last wave field. 
Thu Oct 12 14:12:38, Info:  Writing wave field at time 0.
Thu Oct 12 14:12:38, Info:  Writing wave field at time 0. Done.
Thu Oct 12 14:12:38, Info:  Writing free surface at time 0.
Thu Oct 12 14:12:38, Info:  Writing free surface at time 0. Done.
Thu Oct 12 14:12:38, Info:  Writing energy output at time 0 
Thu Oct 12 14:12:38, Info:  Writing energy output at time 0 Done. 
Thu Oct 12 14:12:41, Info:  Writing free surface at time 0.01.
Thu Oct 12 14:12:41, Info:  Writing free surface at time 0.01. Done.
Thu Oct 12 14:12:42, Info:  Performance since the start: 7232.12 TFLOP/s (rank 0: 406.664 GFLOP/s, average over ranks: 442.873 GFLOP/s) 
Thu Oct 12 14:12:42, Info:  Performance since last sync point: 7232.12 TFLOP/s (rank 0: 406.664 GFLOP/s, average over ranks: 442.873 GFLOP/s) 
```
fixed by #970
